cypress/page-objects/shopPage.js

```diff
@@ -0,0 +1,34 @@
+import BasePage from './basePage';
+import Chance from 'Chance';
+
+const chance = new Chance();
+
+class ShopPage extends BasePage {
+  selectCategory(allCategories) {
+    return chance.pickone(allCategories.collections);
+  }
+
+  selectProduct(selectedCategory) {
+    const selectedItem = chance.pickone(selectedCategory.items);
+    selectedItem.linkUrl = selectedCategory.linkUrl;
+    return selectedItem;
+  }
+
+  pickRandomProduct(listOfProducts) {
+    const selectedCategory = this.selectCategory(listOfProducts);
+    return this.selectProduct(selectedCategory);
+  }
+
+  addProductToCart(pickedItem, quantity = 1) {
+    this.open(pickedItem.linkUrl);
```

Comment:
this line should NOT be here: it is completely different functionality - please do NOT let your methods do too much
so please split this to `navigate to page` and `then add products to cart` steps in your test

Reply:
Sure, done.